File: yizhi/fabric-elytra-autopilot-1.21.7-v3.3.0/src/main/java/net/elytraautopilot/types/FlyToLocation.java
package net.elytraautopilot.types;

import net.elytraautopilot.commands.ClientCommands;
import net.elytraautopilot.config.ModConfig;
import net.elytraautopilot.exceptions.InvalidLocationException;

import static java.lang.Integer.parseInt;

public class FlyToLocation {
    public String Name;
    public int X;
    public int Z;

    public static FlyToLocation ConvertStringToLocation(String configLocation) throws InvalidLocationException {
        String[] tokens = configLocation.split(";");
        if(tokens.length != 3) {
            ModConfig.INSTANCE.flyLocations.remove(configLocation);
            ClientCommands.bufferSave = true;
            throw new InvalidLocationException("Error in reading Fly Location list entry!");
        }
        FlyToLocation location = new FlyToLocation();
        try {
            location.X = parseInt(tokens[1]);
            location.Z = parseInt(tokens[2]);
        } catch (NumberFormatException ignored) {
            ModConfig.INSTANCE.flyLocations.remove(configLocation);
            ClientCommands.bufferSave = true;
            throw new InvalidLocationException("Error in reading Fly Location list entry!");
        }
        location.Name = tokens[0];
        return location;
    }

    public String ConvertLocationToString() {
        return Name + ";" + X + ";" + Z;
    }
}

File: src/main/kotlin/com/zychen027/MeteorPlusPlus/modules/elytraautopilot/types/FlyToLocation.kt
package com.zychen027.meteorplusplus.modules.elytraautopilot.types

import com.zychen027.meteorplusplus.modules.elytraautopilot.exceptions.InvalidLocationException

/**
 * 飞行目标位置 - 来自 ElytraAutoPilot
 * 用于存储命名飞行位置
 */
data class FlyToLocation(
    var name: String = "",
    var x: Int = 0,
    var z: Int = 0
) {
    /**
     * 将位置转换为字符串格式：name;x;z
     */
    fun convertLocationToString(): String {
        return "$name;$x;$z"
    }

    companion object {
        /**
         * 从字符串解析 FlyToLocation
         * 格式：name;x;z
         */
        @Throws(InvalidLocationException::class)
        fun convertStringToLocation(input: String): FlyToLocation {
            val parts = input.split(";")
            if (parts.size != 3) {
                throw InvalidLocationException("Invalid location format: $input")
            }
            
            return FlyToLocation(
                name = parts[0].replace(":", ";"),
                x = parts[1].toInt(),
                z = parts[2].toInt()
            )
        }
    }
}
